Using malloc on ARM causes build failure

I've seen a few queries on the Discord server about using `malloc` for various purposes, such as using `snprintf` to dynamically convert arbitrary value types (`double` or `uint32_t`) into a `send_string`-friendly `char *` string, or declaring a memory allocation to allow injecting Rust code into QMK.

Because Atmel ARM GNU toolchain does not have syscalls enabled, attempting to `make` a keymap with any reference to `malloc` will invariably fail with a linker error similar to the following:

 ```|
 | /usr/lib/gcc/arm-none-eabi/6.3.1/../../../arm-none-eabi/lib/thumb/v7e-m/libg.a(lib_a-sbrkr.o): In function `_sbrk_r':
 | /build/newlib-jo3xW1/newlib-2.4.0.20160527/build/arm-none-eabi/thumb/v7e-m/newlib/libc/reent/../../../../../../../newlib/libc/reent/sbrkr.c:58: undefined reference to `_sbrk'
 | collect2: error: ld returned 1 exit status
 ```
A workaround I've found involves adding the CFLAGs `"-lc -specs=nosys.specs"` to the `make` invocation -- `make CFLAGs+="-lc -specs=nosys.specs" <keyboard>:<keymap>`.  More about this workaround can be read [here](http://ww1.microchip.com/downloads/en/DeviceDoc/Frequently-Asked-Questions-4.9.3.26.txt)

**Caveat emptor** as if you aren't `free`ing up your memory ASAP after `malloc` I could see the board getting into a super sad state.  Be careful!

Opening an issue on this as @yanfali suggested adding this to the documentation.  I would like to open a PR to get this added to documentation but just wanted to get this semi-documented for now.

--- a/tmk_core/chibios.mk
+++ b/tmk_core/chibios.mk
@@ -125,7 +125,7 @@ CHIBISRC = $(STARTUPSRC) \
        $(PLATFORMSRC_CONTRIB) \
        $(BOARDSRC) \
        $(STREAMSSRC) \
-	   $(CHIBIOS)/os/various/syscalls.c
+       $(CHIBIOS)/os/various/syscalls.c
 
 # Ensure the ASM files are not subjected to LTO -- it'll strip out interrupt handlers otherwise.
 QUANTUM_LIB_SRC += $(STARTUPASM) $(PORTASM) $(OSALASM)

--- a/tmk_core/chibios.mk
+++ b/tmk_core/chibios.mk
@@ -124,7 +124,8 @@ CHIBISRC = $(STARTUPSRC) \
        $(PLATFORMSRC) \
        $(PLATFORMSRC_CONTRIB) \
        $(BOARDSRC) \
-       $(STREAMSSRC)
+       $(STREAMSSRC) \
+	   $(CHIBIOS)/os/various/syscalls.c
 
 # Ensure the ASM files are not subjected to LTO -- it'll strip out interrupt handlers otherwise.
 QUANTUM_LIB_SRC += $(STARTUPASM) $(PORTASM) $(OSALASM)

--- a/tmk_core/common.mk
+++ b/tmk_core/common.mk
@@ -27,21 +27,11 @@ TMK_COMMON_SRC +=	$(COMMON_DIR)/host.c \
 
 ifeq ($(PLATFORM),AVR)
   TMK_COMMON_SRC += $(PLATFORM_COMMON_DIR)/xprintf.S
-endif
-
-ifeq ($(PLATFORM),CHIBIOS)
+else ifeq ($(PLATFORM),CHIBIOS)
   TMK_COMMON_SRC += $(PLATFORM_COMMON_DIR)/printf.c
-  ifeq ($(strip $(AUTO_SHIFT_ENABLE)), yes)
-    TMK_COMMON_SRC += $(CHIBIOS)/os/various/syscalls.c
-  else ifeq ($(strip $(TERMINAL_ENABLE)), yes)
-    TMK_COMMON_SRC += $(CHIBIOS)/os/various/syscalls.c
-  endif
-endif
-
-ifeq ($(PLATFORM),ARM_ATSAM)
+else ifeq ($(PLATFORM),ARM_ATSAM)
   TMK_COMMON_SRC += $(PLATFORM_COMMON_DIR)/printf.c
 endif
-
 
 # Option modules
 BOOTMAGIC_ENABLE ?= no

